src/storage.h

```
@@ -0,0 +1,253 @@
+  int ready_time(){ return context()->time() - residence_time; }
+
+ protected:
+  ///   @brief adds a material into the incoming commodity inventory
+  ///   @param mat the material to add to the incoming inventory.
+  ///   @throws if there is trouble with pushing to the inventory buffer.
+  void AddMat_(cyclus::Material::Ptr mat);
+
+  /// @brief Move all unprocessed inventory to processing
+  void BeginProcessing_();
+
+  /// @brief Move as many ready resources as allowable into stocks
+  /// @param cap current throughput capacity 
+  void ProcessMat_(double cap);
+
+  /// @brief any ready resources remaining in processing get pushed off to next timestep
+  /// @param time the timestep whose buffer remains unprocessed 
+  // void AdvanceUnconverted_(int time);
+
+  /// @brief move ready resources from processing to ready at a certain time
+  /// @param time the time of interest
+  void ReadyMatl_(int time);
+
+  /* --- Module Members --- */
+
+  #pragma cyclus var {"tooltip":"input commodity",\
+                      "doc":"commodities accepted by this facility",\
+                      "uilabel":"Input Commodities",\
+                      "uitype":["oneormore","incommodity"]}
+  std::vector<std::string> in_commods;
+
+  #pragma cyclus var {"default": [],\
+                      "doc":"preferences for each of the given commodities, in the same order."\
+                      "Defauts to 1 if unspecified",\
+                      "uilabel":"In Commody Preferences"}
+  std::vector<double> in_commod_prefs;
+
+  #pragma cyclus var {"tooltip":"output commodity",\
+                      "doc":"commodity produced by this facility. Multiple commodity tracking is"\
+                      " currently not supported, one output commodity catches all input commodities.",\
+                      "uilabel":"Output Commodities",\
+                      "uitype":["oneormore","outcommodity"]}
+  std::vector<std::string> out_commods;
```

We could throw an exception if they provide more than one - probably a good idea.